src/test-generator/testgeneratorparamsdialog.cpp

```diff
@@ -0,0 +1,117 @@
+/****************************************************************************
+**
+** Copyright (C) 2018 N7 Space sp. z o. o.
+** Contact: http://n7space.com
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include <QPushButton>
+
+#include <projectexplorer/project.h>
+#include <projectexplorer/session.h>
+
+#include "testgeneratorparamsdialog.h"
+#include "testgeneratorparamswidget.h"
+
+#include "parseddatastorage.h"
+
+using namespace Asn1Acn::Internal;
+using namespace Asn1Acn::Internal::TestGenerator;
+
+TestGeneratorParamsDialog::TestGeneratorParamsDialog(TestGeneratorParamsProviderPtr provider,
```

Comment:
is this provider used anywhere else apart from runner? I just wonder whether it is not a little overkill :) (plain TGParams passed to ->run(params) may be good enough)

Reply:
The idea here is to have shared object (TgParamsProvider) between TestGeneratorsDialog and TestGeneratorRunner. Former sets params, latter reads them.